### PR TITLE
Summon a hero

### LINKED ISSUE #71: Summon a hero
Link the gacha logic to the "Summon" button in the Summon screen.

Core requirements:
- When pressing the "Summon" button, the gacha logic should be called. Its output is a single hero unit. The unit prefab should be shown in the scene.

Nice to haves:
- A nice VFX when the unit appears
- When clicking on the unit, it takes you to the "Unit Detail" screen

### PR DESCRIPTION
**Pull request by ncontinanza**
Closes #71 
⚠️ In order to test this PR, you'll need to download DOTween to the ThirdParty directory, from the link that is in the readme.

## Motivation

The Summon scene functionalities were not yet implemented.

## Summary of changes

- Use the Gacha mechanics to summon two kinds of characters: Rare and Epic. 
- Implement the UI required for that.
- Show the summoned character on screen after summoning.
- The character can be rotated by dragging its model, and the character details scene is loaded after clicking on the character model. The ButtonAnimation class solves this logic, allowing to differentiate a click+drag (used to rotate the model) and a simple click (which is used to see the character details)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,6 +51,7 @@ Ensure you have the following dependencies installed:
     - [RPG inventory icons](https://assetstore.unity.com/packages/2d/gui/icons/rpg-inventory-icons-56687#version-current)
     - [Basic RPG Icons](https://assetstore.unity.com/packages/2d/gui/icons/basic-rpg-icons-181301)
     - [Resource Vector Graphics](https://assetstore.unity.com/packages/2d/gui/icons/resource-icons-101998)
+    - [DOTween](https://dotween.demigiant.com/)
```
Comment (lotuuu) [inline README.md:54]: Not a UnityStore asset :P Please write down that it needs to be dragged into ThirdParty and the additional setup steps it requires

File: client/Assets/Scripts/Gacha/AcceptBehaviour.cs [hunk 2]
```diff
@@ -1,25 +1,27 @@
...
+    private GachaManager gachaManager;
 
     public Box box;
 
-    public void SetBoxByName(string box_name){
-        box = boxes.Find(box => box_name == box.name);
-        item_name.text = box_name;
+    public void SetBoxByName(string boxName){
+        box = boxes.Find(box => boxName == box.name);
+        //itemName.text = boxName;
```
Comment (lotuuu) [inline client/Assets/Scripts/Gacha/AcceptBehaviour.cs:21]: Restore this!

File: client/Assets/Scripts/Gacha/GachaManager.cs [hunk 3]
```diff
@@ -26,5 +38,26 @@ private static void AddNewUnit(User user, Character character)
...
+        characterNameContainer.SetActive(true);
+    }
+
+    public void SelectUnit()
+    {
+        this.SelectUnit(currentUnit);
+    }
+    private void SelectUnit(Unit unit) {
+        UnitDetail.SelectUnit(unit);
     }
```
Comment (lotuuu) [inline client/Assets/Scripts/Gacha/GachaManager.cs:62]: This could be only 1 function

**Review by lotuuu — COMMENTED** [review]
Works great and looks amazing! Few code fixes and we're good to go. Also please write down what we discussed about the back button navigation in the PR description :)

File: client/Assets/Scripts/Shared/ButtonAnimations.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    {
+        var touchXDifference = Math.Abs(eventData.position.x - touchStartPosition.x);
+        var touchYDifference = Math.Abs(eventData.position.y - touchStartPosition.y);
+        if (isInsideCard && touchXDifference < MIN_DIFFERENCE && touchYDifference < MIN_DIFFERENCE)
+        {
+            clickEvent.Invoke();
+        }
+    }
+
+    public override void OnPointerExit(PointerEventData eventData)
```
Comment (lotuuu) [inline client/Assets/Scripts/Shared/ButtonAnimations.cs:67]: I thought we were going to remove these

Comment (lotuuu) [reply client/Assets/Scripts/Shared/ButtonAnimations.cs:67]: Turns out this was necessary after all!

**Review by lotuuu — APPROVED** [review]

File: client/Assets/Scripts/Shared/ButtonAnimations.cs (hunk 4 above)
Comment (ncontinanza) [reply client/Assets/Scripts/Shared/ButtonAnimations.cs:67]: Nope, in the end they were necessary for the button click to work properly

**Pull request merged**